pkg/provider/equinix.go

```diff
@@ -0,0 +1,278 @@
+	}
+
+	kubeConfigFiles := []string{
+		filepath.Join(usr.HomeDir, ".kube/config-plural-backup"),
+		// TODO: make file path more robust
+		filepath.Join(currentDir, "../bootstrap/terraform/kube_config_cluster.yaml"),
+	}
+	kubeconfigs := []*clientcmdapi.Config{}
+
+	for _, filename := range kubeConfigFiles {
+		if len(filename) == 0 {
+			// no work to do
+			continue
+		}
+
+		config, err := clientcmd.LoadFromFile(filename)
+
+		if err != nil {
+			return err
+		}
+
+		kubeconfigs = append(kubeconfigs, config)
+	}
+
+	// first merge all of our maps
+	mapConfig := clientcmdapi.NewConfig()
+
+	for _, kubeconfig := range kubeconfigs {
+		mergo.Merge(mapConfig, kubeconfig, mergo.WithOverride)
+	}
+
+	// merge all of the struct values
+	nonMapConfig := clientcmdapi.NewConfig()
+	for i := range kubeconfigs {
+		kubeconfig := kubeconfigs[i]
+		mergo.Merge(nonMapConfig, kubeconfig, mergo.WithOverride)
+	}
+
+	// since values are overwritten, but maps values are not, we can merge the non-map config on top of the map config and
+	// get the values we expect.
+	config := clientcmdapi.NewConfig()
+	mergo.Merge(config, mapConfig, mergo.WithOverride)
+	mergo.Merge(config, nonMapConfig, mergo.WithOverride)
+
+	json, err := runtime.Encode(clientcmdlatest.Codec, config)
+	if err != nil {
+		return err
+	}
+	output, err := yaml.JSONToYAML(json)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filepath.Join(usr.HomeDir, ".kube/config"), output, 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (equinix *EQUINIXProvider) Install() (err error) {
+	if exists, _ := utils.Which("metal"); exists {
+		utils.Success("metal cli already installed!\n")
+		return
+	}
+
+	fmt.Println("Equinix Metal requires you to manually pkg install the metal cli")
+
+	fmt.Println("Visit https://github.com/equinix/metal-cli#installation to install")
+	return
+}
+
+func (equinix *EQUINIXProvider) Name() string {
+	return EQUINIX
+}
+
+func (equinix *EQUINIXProvider) Cluster() string {
+	return equinix.Clust
+}
+
+func (equinix *EQUINIXProvider) Project() string {
+	return equinix.Proj
+}
+
+func (equinix *EQUINIXProvider) Bucket() string {
+	return equinix.bucket
+}
+
+func (equinix *EQUINIXProvider) Region() string {
+	return equinix.Metro
+}
+
+func (equinix *EQUINIXProvider) Context() map[string]interface{} {
+	return map[string]interface{}{}
+}
+
+func (prov *EQUINIXProvider) Decommision(node *v1.Node) error {
+	// TODO: Figure out how to get and store API token
+	client, err := metal.NewClient()
+
+	if err != nil {
+		return utils.ErrorWrap(err, "Failed to create Equinix Metal client")
+	}
+
+	deviceID := strings.Replace(node.Spec.ProviderID, "equinixmetal://", "", -1)
+
+	_, err = client.Devices.Delete(deviceID, false)
+
+	if err != nil {
+		return utils.ErrorWrap(err, "failed to terminate instance")
+	}
+
+	return nil
+}
+
+func getProjectIDFromName(projectName string) (string, error) {
+	cmd := exec.Command("metal", "project", "get", "-o", "json")
```

Comment:
Is it possible to get this from the go sdk so we can remove dependency on the metal cli?

There's a backlog task to do that for the other providers, but it's more difficult there since it's somewhat required for fetching access tokens in local kubeconfigs.

Reply:
That's probably possible.
The `Decommision` function also doesn't work, since it needs the API token. What is that function used for?